pkg/lib/config/identity.go

```diff
@@ -277,14 +277,23 @@ var _ = Schema.Add("OAuthSSOConfig", `
 }
 `)
 
-func OAuthSSOProviderConfigSchemaBuilder(builder validation.SchemaBuilder) validation.SchemaBuilder {
+func OAuthSSOProviderConfigSchemaBuilder(providerSchemaBuilder validation.SchemaBuilder) validation.SchemaBuilder {
+	builder := validation.SchemaBuilder{}
 	builder.Properties().
 		Property("alias", validation.SchemaBuilder{}.Type(validation.TypeString).MinLength(1)).
 		Property("modify_disabled", validation.SchemaBuilder{}.Type(validation.TypeBoolean)).
 		Property("create_disabled", validation.SchemaBuilder{}.Type(validation.TypeBoolean)).
 		Property("delete_disabled", validation.SchemaBuilder{}.Type(validation.TypeBoolean)).
 		Property("is_active", validation.SchemaBuilder{}.Type(validation.TypeBoolean))
 	builder.AddRequired("alias")
+
+	_if := validation.SchemaBuilder{}
+	_if.Properties().
+		Property("is_active", validation.SchemaBuilder{}.Const(false))
+
+	builder.AllOf(validation.SchemaBuilder{}.If(_if).
+		Else(providerSchemaBuilder))
+
```

Comment:
Add tests for this change. Some cases to test

1. `disabled` is absent, and the original validation should be enforced.
2. `disabled` is `false`, and the original validation should be enforced.
3. `disabled` is `true`, and the original validation should be skipped.

Reply:
Test cases in config_tests already cover 1. I've already added `oauth-provider-inactive` for 3. So I will add one more test case for 2.